beoordelingstool_dockwidget.py

```diff
@@ -57,10 +130,722 @@ def add_herstelmaatregelen(self):
+                if data[JSON_KEY_PROJ][JSON_KEY_NAME]:
+                    json_project[JSON_KEY_NAME] = data[JSON_KEY_PROJ][JSON_KEY_NAME]
+                else:
+                    iface.messageBar().pushMessage("Error", "No project name found.", level=QgsMessageBar.CRITICAL, duration=0)
+                    return json_
+                if data[JSON_KEY_PROJ][JSON_KEY_USERNAME]:
+                    json_project[JSON_KEY_USERNAME] = data[JSON_KEY_PROJ][JSON_KEY_USERNAME]
+                else:
+                    iface.messageBar().pushMessage("Error", "No username found.", level=QgsMessageBar.CRITICAL, duration=0)
+                    return json_
+                if data[JSON_KEY_PROJ][JSON_KEY_URL]:
+                    json_project[JSON_KEY_URL] = data[JSON_KEY_PROJ][JSON_KEY_URL]
+                else:
+                    iface.messageBar().pushMessage("Error", "No project url found.", level=QgsMessageBar.CRITICAL, duration=0)
+                    return json_
+                if data[JSON_KEY_PROJ][JSON_KEY_URL_JSON]:
+                    json_project[JSON_KEY_URL_JSON] = data[JSON_KEY_PROJ][JSON_KEY_URL_JSON]
+                else:
+                    iface.messageBar().pushMessage("Error", "No upload url found for the JSON.", level=QgsMessageBar.CRITICAL, duration=0)
+                    return json_
+                if data[JSON_KEY_PROJ][JSON_KEY_URL_ZIP]:
+                    json_project[JSON_KEY_URL_ZIP] = data[JSON_KEY_PROJ][JSON_KEY_URL_ZIP]
+                else:
+                    iface.messageBar().pushMessage("Error", "No upload url found for the zip-file.", level=QgsMessageBar.CRITICAL, duration=0)
+                    return json_
+            else:
+                iface.messageBar().pushMessage("Error", "No json found.", level=QgsMessageBar.CRITICAL, duration=0)
+                return
+            json_[JSON_KEY_PROJ] = json_project
+            json_["pipes"] = pipes
+            json_["manholes"] = manholes
+            with open("{}".format(json_path), 'w') as outfile:
+                json.dump(json_, outfile, indent=2)
+            return json_
+        else:
+            iface.messageBar().pushMessage("Warning", "You don't have a manholes, pipes and measuring_points layer.", level=QgsMessageBar.WARNING, duration=0)
+            json_ = {}
+            return json_
+
+
+    def get_selected_manhole(self):
+        layer = iface.activeLayer()
+        fields = layer.dataProvider().fields()
+        for f in layer.selectedFeatures():
+            self.field_combobox_manholes.setCurrentIndex(self.field_combobox_manholes.findText(str(f["Herstelmaa"]))) if self.field_combobox_manholes.findText(str(f["Herstelmaa"])) else self.field_combobox_manholes.setCurrentIndex(0)
+            self.value_plaintextedit_manholes.setPlainText(f["Opmerking"] if type(f["Opmerking"]) is not QPyNullVariant else "")
+            self.tablewidget_manholes.setItem(0, 0, QTableWidgetItem(f["CAA"]))
```

Comment:
Nu niet veranderen, maar... Dit kan je ook met een lijst of een dict doen. 

    mapping = [[1, 'CAJ'], [2, 'CAL'], ...]
    for row_nr, field in mapping:
        self.tablewidget_....setItem(0, row_nr, QTabl....(f[field]))